src/v/cloud_storage/remote_partition.cc

```diff
@@ -752,6 +752,149 @@ ss::future<bool> remote_partition::tolerant_delete_object(
     }
 }
 
+ss::future<bool> remote_partition::tolerant_delete_objects(
+  const cloud_storage_clients::bucket_name& bucket,
+  std::vector<cloud_storage_clients::object_key>&& keys,
+  retry_chain_node& parent) {
+    if (keys.empty()) {
+        co_return false;
+    }
+
+    // Copy first key so that we can use it in error messages
+    auto first_key = keys[0];
+
+    auto result = co_await _api.delete_objects(bucket, std::move(keys), parent);
+    if (result == upload_result::timedout) {
+        throw std::runtime_error(fmt::format(
+          "Timed out deleting objects for partition {} (keys {}-)",
+          _manifest.get_ntp(),
+          first_key));
+    } else if (result != upload_result::success) {
+        vlog(
+          _ctxlog.warn,
+          "Error ({}) deleting objects for partition (keys {}-)",
+          result,
+          first_key);
+        co_return true;
+    } else {
+        co_return false;
+    }
+}
+
+static constexpr ss::lowres_clock::duration erase_timeout = 60s;
+static constexpr ss::lowres_clock::duration erase_backoff = 1s;
+static constexpr ss::lowres_clock::duration erase_non_0th_delay = 200ms;
+
+/**
+ * Return the index of this node in the list of voters, or nullopt if it
+ * is not a voter.
+ */
+std::optional<size_t>
+voter_position(raft::vnode self, const raft::group_configuration& raft_config) {
+    const auto& voters = raft_config.current_config().voters;
+    auto position = std::find(voters.begin(), voters.end(), self);
+    if (position == voters.end()) {
+        return std::nullopt;
+    } else {
+        return position - voters.begin();
+    }
+}
+
+ss::future<remote_partition::finalize_result> remote_partition::finalize(
+  ss::abort_source& as,
+  raft::vnode self,
+  raft::group_configuration raft_config) {
+    vlog(_ctxlog.info, "Finalizing remote storage state...");
+
+    // To reduce redundant re-uploads in the typical case of all replicas
+    // being alive, have all non-0th replicas delay before attempting to
+    // reconcile the manifest. This is just a best-effort thing, it is
+    // still okay for them to step on each other: finalization is best
+    // effort and the worst case outcome is to leave behind a few orphan
+    // objects if writes were ongoing while deletion happened.
+    auto my_position = voter_position(self, raft_config);
+    if (my_position.has_value()) {
+        auto p = my_position.value();
+        if (p != 0) {
+            co_await ss::sleep_abortable(erase_non_0th_delay * p, as);
```

Comment:
nit: If deletion completes successfully on the replica 0, then, when the other replicas wake up they'll fail to download the manifest and produce warning logs. Perhaps we should keep a flag for this case and silence the log?

Reply:
Good point... I think this is also the existing behavior, but yeah, we should fix it.

I think it makes sense to fix this up along with some tombstone stuff, to make the situation where a manifest is 404 less ambiguous.